### PR TITLE
added options to configure ssl support for logstash

### PR DESCRIPTION
**Pull request by ricardokirkner**
Allow specifying ssl cert/key to encrypt communicatio between filebeat and logstash.

### CODE REVIEW

**Comment by ricardokirkner** [comment]
I was actually expecting that the values are provided in base64 form. Maybe it would be best to make that explicit in the description of the option?

**Comment by chuckbutler** [comment]
Ok, without having run the code:  Does the render function somehow expand this from base64 encoded format?

To me it looks like its just going to whole-sale dump the base64 decoded data on disk. pardon me if i missed something or misinterpreted it :)

**Comment by ricardokirkner** [comment]
Yes! sorry ... I was in the k8s mindset where secrets are shipped base64 encoded... and slipped this obvious "detail"... In the latest change I pushed the ssl cert data is now expected to be provided as a bas64 encoded string and will be decoded when writing to disk.

File: reactive/filebeat.py [hunk 1]
```diff
@@ -32,6 +38,29 @@ def render_filebeat_template():
         status_set('active', 'Filebeat ready.')
 
 
+@restart_on_change({
+    LOGSTASH_SSL_CERT: ['filebeat'],
+    LOGSTASH_SSL_KEY: ['filebeat'],
+    })
+@when('beat.render')
+@when('apt.installed.filebeat')
```
Comment (Cynerva) [inline reactive/filebeat.py]: These `@when` decorators need to go above the `@restart_on_change`. As it is now, the `@when` decorators will have charms.reactive calling the unwrapped `render_filebeat_logstash_ssl_cert`, not the wrapped version created by `@restart_on_change`.

**Pull request merged**